jdaviz/configs/default/plugins/collapse/collapse.py

```diff
@@ -148,28 +125,26 @@ def vue_collapse(self, *args, **kwargs):
         # Link the new dataset pixel-wise to the original dataset. In general
         # direct pixel to pixel links are the most efficient and should be
         # used in cases like this where we know there is a 1-to-1 mapping of
-        # pixel coordinates. Here which axes are linked to which depends on
-        # the selected axis.
-        (i1, i2), (i1c, i2c) = AXES_MAPPING[self.selected_axis]
+        # pixel coordinates.
+        # Spatial-spatial image only.
+        i1, i2, i1c, i2c = 0, 1, 0, 1
 
         pix_id_1 = self._selected_data.pixel_component_ids[i1]
         pix_id_1c = self.data_collection[label].pixel_component_ids[i1c]
         pix_id_2 = self._selected_data.pixel_component_ids[i2]
         pix_id_2c = self.data_collection[label].pixel_component_ids[i2c]
 
-        self.data_collection.add_link(LinkSame(pix_id_1, pix_id_1c))
-        self.data_collection.add_link(LinkSame(pix_id_2, pix_id_2c))
+        self.data_collection.add_link([LinkSame(pix_id_1, pix_id_1c),
+                                       LinkSame(pix_id_2, pix_id_2c)])
 
         snackbar_message = SnackbarMessage(
             f"Data set '{self._selected_data.label}' collapsed successfully.",
             color="success",
             sender=self)
         self.hub.broadcast(snackbar_message)
 
-        if self.selected_axis == 0 and self.selected_viewer != 'None':
+        # Spatial-spatial image only.
+        if self.selected_viewer != 'None':
             # replace the contents in the selected viewer with the results from this plugin
             self.app.add_data_to_viewer(self.viewer_to_id.get(self.selected_viewer),
```

Comment:
@kecnry , I noticed something weird where while the collapsed image shows in the selected viewer, I am unable to use Layers menu to change the display. I have to deselect it from Data and re-select for Layers to work again. Not sure if this is a known issue or not.

Also, the spectral Subset somehow colors the whole collapsed image in the tint of the Subset color. I can disable it by clicking the eye icon on the Layers dropdown. I think this is a separate known issue already.